Improve exception logging during startup

arjantijms opened the pull request:
Some exceptions would not get logged during startup, particularly multi exceptions from HK2 wrapped in other exceptions and/or being suppressed.

Signed-off-by: Arjan Tijms <arjan.tijms@gmail.com>

arjantijms committed:
Improve exception logging during startup

Signed-off-by: Arjan Tijms <arjan.tijms@gmail.com>

dmatej commented on nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/osgi/EmbeddedOSGiGlassFishRuntime.java:
```diff
@@ -144,20 +156,41 @@ public synchronized GlassFish newGlassFish(GlassFishProperties gfProps) throws G
+            for (Throwable suppressed : currentThrowable.getSuppressed()) {
+                MultiException multiException = findMultiException(suppressed);
+                if (multiException != null) {
+                    return multiException;
```
But this way you lose other possible suppressed exceptions describing what happened, right?
I need some more detailed explanation (javadoc), at least why do you think it is ok. And comparison of the output to see it too.

arjantijms replied:
> But this way you lose other possible suppressed exceptions describing what happened

I don't think so, as the original exception (throwable t) is still propagated:

```
throw new GlassFishException(bundleBuilder.toString(), t);
```

The first `MultiException` encountered is just used for `bundleBuilder.toString`.

dmatej replied:
Oh, I see, thanks!

dmatej commented on nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/ASMain.java:
```diff
@@ -17,16 +17,26 @@
+        try {
+            GlassFishMain.main(args);
+        } catch (Throwable t) {
+            LOG.log(Level.SEVERE, "Error starting GlassFish", t);
```
If GF fails to start, this cannot print anyhing, because GlassFish will not load logging.properties. So in this case t.printStacktrace() would be better.

arjantijms replied:
It does actually gets added to the output (not the log file). So you do see what happened if GF doesn't start there. Of course, we could do `t.printStacktrace()` if that's better?

dmatej committed:
Fixed tests - cannot use JUL before we try to set the log manager

- it would not even work, if gf cannot start, it cannot load it's logging,
  so logs would be lost.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>
